Client/Application/AppDelegate.swift

```diff
@@ -135,6 +142,52 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
         SDWebImageDownloader.sharedDownloader().setValue(firefoxUA, forHTTPHeaderField: "User-Agent")
     }
+
+    func application(application: UIApplication, handleActionWithIdentifier identifier: String?, forLocalNotification notification: UILocalNotification, completionHandler: () -> Void) {
+        if let actionId = identifier {
+            if let action = SentTabAction(rawValue: actionId) {
+                switch(action) {
+                case .View:
+                    viewURLInNewTab(notification)
+                    break
+                case .Bookmark:
+                    addBookmark(notification)
+                    break
+                case .ReadingList:
+                    addToReadingList(notification)
+                    break
+                }
+            } else {
+                println("ERROR: Unknown notification action received")
+            }
+        } else {
+            println("ERROR: Unknown notification received")
+        }
+    }
+
+    func application(application: UIApplication, didReceiveLocalNotification notification: UILocalNotification) {
+        viewURLInNewTab(notification)
+    }
+
+    private func viewURLInNewTab(notification: UILocalNotification) {
+        if let alertURL = notification.userInfo?[TabSendURLKey] as? String{
+            browserViewController.openURLInNewTab(NSURL(string: alertURL)!)
```

Comment:
Same here. Every time you force unwrap a value, be really really sure that it is fail safe. In this case it is not because `alertURL` could be malformed. Which means we crash.